### PR TITLE
#572: Added support for LLEN command

### PR DESCRIPTION
**Pull request by jainbhavya53**
Closes https://github.com/DiceDB/dice/issues/572

# Changes
1. Added support for LLEN
2. Added unit tests and integration tests

# Test Results
All tests passed.

### CODE REVIEW

**Comment by jainbhavya53** [comment]
@JyotinderSingh Can you please review it ?
Thanks

File: internal/eval/eval.go [hunk 1]
```diff
@@ -2219,6 +2219,33 @@ func evalLPOP(args []string, store *dstore.Store) []byte {
...
+
+	if err := object.AssertType(obj.TypeEncoding, object.ObjTypeByteList); err != nil {
+		return clientio.Encode(err, false)
+	}
+
+	if err := object.AssertEncoding(obj.TypeEncoding, object.ObjEncodingDeque); err != nil {
+		return clientio.Encode(err, false)
+	}
+
+	deq := obj.Value.(*Deque)
```
Comment (JyotinderSingh) [inline internal/eval/eval.go:2245]: Can't we simply use 
```go
object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeByteList, object.ObjEncodingDeque)
```

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for adding support for LLEN command, @jainbhavya53! The changes look good overall, left a small comment.

File: internal/eval/eval.go (hunk 1 above)
Comment (jainbhavya53) [reply internal/eval/eval.go:2245]: sure let me make the change

Comment (jainbhavya53) [reply internal/eval/eval.go]: Done

**Comment by jainbhavya53** [comment]
Rebased the changes over master to resolve merge conflicts
Addressed Review Comments

File: internal/eval/eval.go [hunk 2]
```diff
@@ -2227,6 +2227,29 @@ func evalLPOP(args []string, store *dstore.Store) []byte {
...
+		return clientio.Encode(0, false)
+	}
+
+	// if object is a set type, return error
+	if object.AssertType(obj.TypeEncoding, object.ObjTypeSet) == nil {
+		return diceerrors.NewErrWithFormattedMessage(diceerrors.WrongTypeErr)
+	}
+
+	if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeByteList, object.ObjEncodingDeque); err != nil {
+		return clientio.Encode(err, false)
```
Comment (JyotinderSingh) [inline internal/eval/eval.go:2246]: Return the error directly, it is already encoded. Include a test case validating the error message too.
```suggestion
		return err

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]

File: internal/eval/eval.go [hunk 3]
```diff
@@ -2227,6 +2227,29 @@ func evalLPOP(args []string, store *dstore.Store) []byte {
...
+
+	obj := store.Get(args[0])
+	if obj == nil {
+		return clientio.Encode(0, false)
+	}
+
+	// if object is a set type, return error
+	if object.AssertType(obj.TypeEncoding, object.ObjTypeSet) == nil {
+		return diceerrors.NewErrWithFormattedMessage(diceerrors.WrongTypeErr)
+	}
```
[the remark this answers is not in the document]
Comment (jainbhavya53) [reply internal/eval/eval.go]: done

File: internal/eval/eval.go (hunk 2 above)
Comment (jainbhavya53) [reply internal/eval/eval.go]: Done

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for the LLEN command, @jainbhavya53! LGTM.

**Pull request merged**